x-pack/platform/plugins/shared/onechat/server/services/tools/persisted/converter.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ToolTypeDefinition } from '../tool_types';
+import type { InternalToolDefinition } from '../tool_provider';
+import type { ToolPersistedDefinition } from './client';
+import type { ToolTypeConversionContext } from '../tool_types/definitions';
+
+export const convertPersistedDefinition = ({
+  tool,
+  definition,
+  context,
+}: {
+  tool: ToolPersistedDefinition;
+  definition: ToolTypeDefinition;
+  context: ToolTypeConversionContext;
+}): InternalToolDefinition => {
+  const { id, type, description, tags, configuration } = tool;
+  const { request, spaceId } = context;
+
+  const getDynamicProps = () => {
+    return definition.getDynamicProps(configuration, { request, spaceId });
+  };
+
+  return {
+    id,
+    type,
+    description,
+    tags,
+    configuration,
+    readonly: false,
+    getSchema: async () => {
+      const props = await getDynamicProps();
+      return props.getSchema();
+    },
+    getHandler: async () => {
+      const props = await getDynamicProps();
+      return props.getHandler();
+    },
+    getLlmDescription: async (args) => {
+      const props = await getDynamicProps();
+      return props.getLlmDescription ? props.getLlmDescription(args) : description;
+    },
```

Comment:
So that double layer of accessors (`getDynamicProps` on the type def split on the actual tool instances) may look awkward, but the idea is being able to add caching later for tool types requiring data access (thinking workflow tool mostly - we need the workflow definition to generate the schema or description)